token/js/src/instructions/amountToUiAmount.ts

```diff
@@ -0,0 +1,124 @@
+import { struct, u8} from '@solana/buffer-layout';
+import { u64 } from '@solana/buffer-layout-utils';
+import { AccountMeta, PublicKey, TransactionInstruction } from '@solana/web3.js';
+import { TOKEN_PROGRAM_ID } from '../constants';
+import {
+    TokenInvalidInstructionDataError,
+    TokenInvalidInstructionKeysError,
+    TokenInvalidInstructionProgramError,
+    TokenInvalidInstructionTypeError,
+} from '../errors';
+import { TokenInstruction } from './types';
+
+/** TODO: docs */
+export interface AmountToUiAmountInstructionData {
+    instruction: TokenInstruction.AmountToUiAmount;
+    amount: bigint;
+}
+
+/** TODO: docs */
+export const amountToUiAmountInstructionData = struct<AmountToUiAmountInstructionData>([u8('instruction'), u64('amount')]);
+
+/**
+ * Construct a AmountToUiAmount instruction
+ *
+ * @param mint         Public key of the mint
+ * @param amount       Amount of tokens to be converted to UiAmount
+ * @param programId    SPL Token program account
+ *
+ * @return Instruction to add to a transaction
+ */
+export function createAmountToUiAmountInstruction(
+    mint: PublicKey,
+    amount: number | bigint,
+    programId = TOKEN_PROGRAM_ID
+): TransactionInstruction {
+    const keys = [{ pubkey: mint, isSigner: false, isWritable: true }];
```

Comment:
Mint isn't writable in this instruction either
```suggestion
    const keys = [{ pubkey: mint, isSigner: false, isWritable: false }];
```